exe/node/Main.hs

```diff
@@ -115,40 +115,55 @@ main = do
-    forkEkgServer metricsStore ekgHost ekgPort
-
-    withStdLogger (Log.configForEnvironment environment) $ \lgr -> Log.withExceptionLogged lgr $
-        BlockStore.Concrete.STM.withBlockStore (fromGenesis gen) Nakamoto.blockScore $ \blkStore@(publicBlockStore, _) -> do
-            let telemetryHandle = Telemetry.newTelemetryStore lgr metricsStore
-            runProtocol Nakamoto.validateFull Nakamoto.blockScore telemetryHandle blkStore config $ \proto ->
-                withNode (mkNodeConfig environment telemetryHandle noEmptyBlocks config)
-                         nid
-                         mem
-                         stStore
-                         publicBlockStore
-                         proto
-                         Rad.txEval
-                         consensus                                      $ \nod ->
-                    withGossip telemetryHandle
-                               P2P.NodeAddr { P2P.nodeId   = pure nid
-                                            , P2P.nodeHost = host
-                                            , P2P.nodePort = gossipPort
-                                            }
-                               seeds
-                               (storage nod)
-                               (Handshake.simpleHandshake keys)         $ \gos ->
-                        Async.runConcurrently $
-                                 Async.Concurrently (HTTP.run (fromIntegral apiPort) nod)
-                              <> Async.Concurrently (miner nod gos)
+    let consensusConfig = Consensus.getConfig environment
+
+    runManaged $ do
+        lgr <- managed $ withStdLogger (Log.configForEnvironment environment)
+
+        let telemetry = Telemetry.newTelemetryStore lgr metricsStore
+
+        blkStore@(blkStoreReader,_) <- managed $
+            BlockStore.STM.withBlockStore (fromGenesis gen) Nakamoto.blockScore
+
+        proto <- managed $
+            runProtocol Nakamoto.validateFull
+                        Nakamoto.blockScore
+                        telemetry
+                        blkStore
+                        consensusConfig
+
+        node <- managed $
+            withNode (mkNodeConfig environment telemetry noEmptyBlocks consensusConfig)
+                     nid
+                     mem
+                     stStore
+                     blkStoreReader
+                     proto
+                     Rad.txEval
+                     consensus
+
+        gossip <- managed $
+            withGossip telemetry
+                       P2P.NodeAddr { P2P.nodeId   = pure nid
+                                    , P2P.nodeHost = host
+                                    , P2P.nodePort = gossipPort
+                                    }
+                       seeds
+                       (storage node)
+                       (Handshake.simpleHandshake keys)
+
+        liftIO . Log.withExceptionLogged lgr $ do
+            forkEkgServer metricsStore ekgHost ekgPort
```

Comment:
Just to make sure my telemetry code relied in some crazy ordering of event -- are you still able to run the node and open the EKG dashboard? (the `--help` should tell you which port EKG listens by default).

Reply:
Yeah this works. It would be preferable to be able to manage the ekg thread in `main`, and make sure it starts up _only_ after the other services have started successfully. But I'll leave that for another time :)

Reply:
Right, I was dwelling upon that as well myself, and wrote a half-baked comment which I scrapped which essentially was wondering if we should have called that function `runEkgServer` and then make it run in the `runConcurrently` block with the rest, for consistency.

Problem being we piggyback on EKG's `forkServerWith`, and _I think_ `ekg-core` itself doesn't expose the relevant function to start up the ekg server synchronously.